src/RevisionPopupWidget.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @class
+ * @constructor
+ */
+const RevisionPopupWidget = function RevisionPopupWidget() {
+	// Parent constructor
+	RevisionPopupWidget.parent.call( this, {
+		padded: true,
+		autoClose: true,
+		position: 'above',
+		// FIXME: 'force-left' for RTL languages
+		align: 'force-right',
+		hideWhenOutOfView: false,
+		$content: $( '<div>' )
+			.addClass( 'ext-wwt-revisionPopupWidget-content' )
+	} );
+};
+
+/* Setup */
+OO.inheritClass( RevisionPopupWidget, OO.ui.PopupWidget );
+
+/**
+ * Update the contents of the revision popup.
+ * @param {Object} data As returned by Api.getTokenInfo().
+ */
+RevisionPopupWidget.prototype.setContent = function ( data ) {
+	// Note that the English namespaces will normalize to the wiki's local namespaces.
+	const userLinks = `
+		<a target="_blank" href="${mw.util.getUrl( `User:${data.username}` )}">${data.username}</a>
+		${mw.msg( 'parentheses-start' )}<a target="_blank" href="${mw.util.getUrl( `User talk:${data.username}` )}">${mw.msg( 'talkpagelinktext' )}</a>
+		${mw.msg( 'pipe-separator' )}
+		<a target="_blank" href="${mw.util.getUrl( `Special:Contributions/${data.username}` )}">${mw.msg( 'contribslink' )}</a>${mw.msg( 'parentheses-end' )}
+	`,
+		// eslint-disable-next-line no-undef
+		dateStr = moment( data.revisionTime ).locale( mw.config.get( 'wgUserLanguage' ) ).format( 'LLL' ),
+		diffLink = `<a target="_blank" href="${mw.util.getUrl( `Special:Diff/${data.revisionId}` )}">${dateStr}</a>`,
+		addedMsg = mw.message( 'ext-whowrotethat-revision-added', userLinks, diffLink ).parse(),
+		attributionMsg = mw.message( 'ext-whowrotethat-revision-attribution', data.score ).parse();
+
+	$( '.ext-wwt-revisionPopupWidget-content' ).html( `${addedMsg.trim()} ${attributionMsg}` );
```

Comment:
I know that we're the ones setting this data, but there still might be some unpredictable input; I'd use `$.parseHTML()` here for safety.